File: 602_BaseBall_Game.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        z=[]
        for i in ops:
            if i!="C" and i!="+" and i!="D":
                z.append(int(i))
            if i=="+":
                z.append(int(z[-1])+int(z[-2]))
            elif i=="D":
                z.append(2*int(z[-1]))
            elif i=="C":
                z.pop()
        return sum(z)

File: 602_BaseBall_Game.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int> q;
        for(string s : ops){
            switch(s[0]){
                case 'C':
					if(q.empty()) break;
                    q.pop_back(); break;
                case '+':
                    switch(q.size()){
                        case 0: break;
                        case 1: q.emplace_back(q.back()); break;
                        default:
                            auto itend = q.end() - 1;
                            q.emplace_back(*itend + *(itend - 1));
                            break;
                    } break;
                case 'D':
                    if(q.empty()) break;
                    q.emplace_back(q.back() * 2); break;
                default :
                    q.emplace_back(std::stoi(s)); break;
            }
        }        
        int sum = 0;
        for(int x : q){
            sum += x;
        }
        return sum;
    }
};